src/hooks/Provider.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { createContext, useState, useContext } from 'react';
+import { fetchRequest } from '../services/fetchRequest';
+
+const RestyContext = createContext();
+
+export const RestyProvider = ({ children }) => {
```

Comment:
Don't forget to use prop-types for any props you pass to your components. For children you want to use PropTypes.nodes like this:
```js
RestyProvider.propTypes = {
  children: PropTypes.node
};
```